### PR TITLE
[MOB-2121] get in-apps and register for push after a new token

### PR DESCRIPTION
**Pull request by roninopf**

### CODE REVIEW

File: swift-sdk/Internal/AuthManager.swift [hunk 1]
```diff
@@ -13,14 +13,20 @@ import Foundation
 
 class AuthManager: IterableInternalAuthManagerProtocol {
     init(onAuthTokenRequestedCallback: (() -> String?)?,
+         autoPushRegistration: Bool,
          localStorage: LocalStorageProtocol,
          dateProvider: DateProviderProtocol,
+         notificationStateProvider: NotificationStateProviderProtocol,
+         inAppManager: IterableInternalInAppManagerProtocol,
```
Comment (tapashmajumder) [inline swift-sdk/Internal/AuthManager.swift:20]: The design looks strange. AuthManager should not have a dependency on inAppManager.

Comment (roninopf) [reply swift-sdk/Internal/AuthManager.swift:20]: The issue arises from the need to reset in-apps and registration. I'll figure out a new design for it.

File: swift-sdk/Internal/IterableAPIInternal.swift [hunk 2]
```diff
@@ -121,7 +122,9 @@ final class IterableAPIInternal: NSObject, PushTrackerProtocol, AuthProvider {
             _email = email
             _userId = nil
             
-            authManager.requestNewAuthToken(hasFailedPriorAuth: false)
+            authManager.requestNewAuthToken(hasFailedPriorAuth: false) {
+                self.loginNewUser()
```
Comment (tapashmajumder) [inline swift-sdk/Internal/IterableAPIInternal.swift:126]: This will cause a memory leak.

Comment (tapashmajumder) [inline swift-sdk/Internal/IterableAPIInternal.swift:126]:
```suggestion
            authManager.requestNewAuthToken(hasFailedPriorAuth: false) { [weak self] in
                self?.loginNewUser()
```
```suggestion
            authManager.requestNewAuthToken(hasFailedPriorAuth: false) {
                self.loginNewUser()
```

File: swift-sdk/Internal/DependencyContainer.swift [hunk 3]
```diff
@@ -41,7 +41,8 @@ extension DependencyContainerProtocol {
                      retryInterval: config.inAppDisplayInterval)
     }
     
-    func createAuthManager(config: IterableConfig) -> IterableInternalAuthManagerProtocol {
+    func createAuthManager(config: IterableConfig,
+                           inAppManager: IterableInternalInAppManagerProtocol) -> IterableInternalAuthManagerProtocol {
```
Comment (tapashmajumder) [inline swift-sdk/Internal/DependencyContainer.swift:45]: inAppManager should not be a dependency here.

File: swift-sdk/Internal/IterableAPIInternal.swift [hunk 4]
```diff
@@ -68,7 +68,8 @@ final class IterableAPIInternal: NSObject, PushTrackerProtocol, AuthProvider {
     }()
     
     lazy var authManager: IterableInternalAuthManagerProtocol = {
-        self.dependencyContainer.createAuthManager(config: self.config)
+        self.dependencyContainer.createAuthManager(config: self.config,
+                                                   inAppManager: self.inAppManager)
```
Comment (tapashmajumder) [inline swift-sdk/Internal/IterableAPIInternal.swift:72]: No need for in-app manager

**Review by tapashmajumder — APPROVED** [review]
+1

**Pull request merged**